lidar_topic: "/laser_front/scan_virtual"
# goal: "move_base_simple/goal"
# path: "path"
cmd_vel: "cmd_vel" # Private topic named cmd_vel

driver:
  local:
    critical_radius: 0.9 #m
    safe_radius: 3 #m
    max_linear_velocity: 1 #m/S
    max_angular_velocity: 0.5 #rad/s
    velocity_gain: 1.2 
    angular_velocity_gain: 1.6 
    angular_slowdown_gain: 0.8 
    

# Costmaps parameters
map:
  global:
    # Plugins
    plugins:
      # - {name: "negative_obstacles_layer", type: "void_detection::NegativeObstacleLayer"}
      - { name: "obstacles_layer", type: "costmap_2d::ObstacleLayer" }
      - { name: "inflation_layer", type: "costmap_2d::InflationLayer" }
    footprint: [[0.85,0.6],[0.85,-0.6],[-0.7,-0.6],[-0.7,0.6]]
    # Coordinate frame and tf parameters
    global_frame: "world"
    robot_base_frame: "base_footprint"
    # Rate parameters
    update_frequency: 1.0 # Hz
    publish_frequency: 1.0 # Hz (default: 0 Hz)
    # Map management parameters
    rolling_window: true
    # The following parameters can be overwritten by some layers, namely the static map layer.
    width: 40 # m
    height: 40 # m
    resolution: 0.2 # m/cell
    # Plugins configurations
    obstacles_layer:
      observation_sources: "LiDAR"
      LiDAR:
        topic: "laser_front/scan_virtual"
        sensor_frame: "velodyne"
        data_type: "LaserScan"
        clearing: true
        marking: true
        # Global filtering parameters
        obstacle_range: 20 # m (default: 2.5)
        raytrace_range: 20 # m (default: 3.0)
    inflation_layer:
      inflation_radius: 1.5 # m
      cost_scaling_factor: 2.5
    negative_obstacles_layer:
      depth_camera_topic: "/depth_camera/depth/image_raw"
      horizontal_fov: 60 # degrees
      vertical_fov: 60 # degrees
  local:
    # Plugins
    plugins:
      - { name: "obstacles_layer", type: "costmap_2d::ObstacleLayer" }
      - { name: "inflation_layer", type: "costmap_2d::InflationLayer" }
    footprint: [[0.85,0.6],[0.85,-0.6],[-0.7,-0.6],[-0.7,0.6]]
    # Coordinate frame and tf parameters
    global_frame: "world"
    robot_base_frame: "base_footprint"
    # Rate parameters
    update_frequency: 15.0 # Hz
    publish_frequency: 1.0 # Hz (default: 0 Hz)
    # Map management parameters
    rolling_window: true
    # The following parameters can be overwritten by some layers, namely the static map layer.
    width: 10 # m
    height: 10 # m
    resolution: 0.1 # m/cell
    # Plugins configurations
    obstacles_layer:
      observation_sources: "LiDAR"
      LiDAR:
        topic: "laser_front/scan_virtual"
        sensor_frame: "velodyne"
        data_type: "LaserScan"
        clearing: true
        marking: true
        # Global filtering parameters
        obstacle_range: 10 # m (default: 2.5)
        raytrace_range: 10 # m (default: 3.0)
    inflation_layer:
      inflation_radius: 1.5 # m
      cost_scaling_factor: 2.0
